pkg/project/auth/watch.go

```diff
@@ -158,16 +173,26 @@ func (w *userProjectWatcher) Watch() {
 	}()
 	defer utilruntime.HandleCrash()
 
+	// start by emitting all the `initialProjects`
+	for i := range w.initialProjects {
+		// keep this check here to sure we don't keep this open in the case of failures
+		select {
+		case err := <-w.cacheError:
+			w.emit(makeErrorEvent(err))
+			return
+		default:
+		}
+
+		w.emit(watch.Event{
+			Type:   watch.Added,
```

Comment:
> I know this watch isn't normal but is this what happens in normal watches? We get only Added events up to the point we start watching?

After it finishes sending the adds, it will drop into the "normal" operation loop below where it will continue the watch after its initial list.

I'm pretty sure that it only gives Add events with the current state of the object instead of every individual add/modify/remove.  @liggitt care to confirm or deny?  Even if it did, that information isn't available here, so its the best we can do.